src/snowflake/connector/network.py

```diff
@@ -187,6 +187,7 @@
 OAUTH_AUTHENTICATOR = "OAUTH"
 ID_TOKEN_AUTHENTICATOR = "ID_TOKEN"
 USR_PWD_MFA_AUTHENTICATOR = "USERNAME_PASSWORD_MFA"
+PROGRAMMATIC_ACCESS_TOKEN = "PROGRAMMATIC_ACCESS_TOKEN"
```

Comment:
How user will provide authentication name in connection parameter? like "PROGRAMMATIC_ACCESS_TOKEN"? If yes, then can we use "PAT" instead of long name?

Reply:
Here's the exact shell command I was using to test the feature: `python -c 'from snowflake.connector import connect; c=connect(account="account", user="user", authenticator="PROGRAMMATIC_ACCESS_TOKEN", token_file_path="/Users/mkeller/pat"); print(c.cursor().execute("select 1, 2").fetchall())'`

I haven't introduced the short form simply because we don't offer short-hands for any of the other authentications. I'm happy to introduce it if you think we should start now though!